### ISSUE TITLE
Better data caching for repeated experiments

### ISSUE THREAD

**Issue by janvainer** [pyannote/pyannote-audio#507]
**Is your feature request related to a problem? Please describe.**
I am finetuning a pretrained SAD model on custom data based on [this tutorial](https://github.com/pyannote/pyannote-audio/tree/master/tutorials/models/speech_activity_detection). When I run training, I get the following message
![obrazek](https://user-images.githubusercontent.com/31015638/99074679-80cc1280-25b8-11eb-9418-12b3c2adfe23.png)
and the code starts loading labels, which takes a very long time even for a relatively small dataset - 400 recordings of 30 minute long audio files. This would be ok if it happened only the first time I try to finetune, but the label loading happens every single time I run the same training with the same data, which is very wasteful.

**Describe the solution you'd like**
I would like to preprocess my data once and then be able to reuse them based on some simple checksum. Ideally, when the training command is executed, the network should start to train within several seconds/minutes and not wait for data processing for over an hour.

I tried to find some caching options in the docs, but had no luck. Is there some option to store precomputed features etc to avoid repeated data preprocessing? Also what exactly happens inside loading labels and how long is it expected to run wrt the length of the training rttm file? Any suggestions would be appreciated. Thanks :)

**Comment by hbredin** [comment]
This issue will be addressed in upcoming v2 which is not ready yet (and I cannot promise any ETA).

That being said, you may want to have a look at `v2` branch and [this notebook](https://github.com/pyannote/pyannote-audio/blob/v2/notebook/example.ipynb) in particular to see what's coming and try it for yourself.

**Comment by LordOfLuck** [comment]
Hi, thanks for response. I will check out the notebook. Regards, Jan

**Comment by LordOfLuck** [comment]
Wow cant wait for v2 @hbredin :) It looks really good so far.

**Comment by LordOfLuck** [comment]
@hbredin I managed to fine-tune the model. Now I want to optimize hyperparams and need to extract raw data. However, the inference takes ages and I am not sure if this is expected or if there is something wrong with my data manifests. I have around 15 validation recordings and I have rttm file that contains speech segments from all the recordings and has some thousands of rows. There is a timer showing `Development: 48file [1:47:35, 153.71s/file]`. What does the number 48 represent? I do not have that many recordings. Is it the 48-th item from the rttm file? 

**How can I speed up the inference? Since there is no need for ground truth VAD, would it be more efficient to have one item per recording in the rttm file?** I suspect that the inference right now happens for each item in the rttm separately and running inference for each recording as a whole might be faster, but I would like to be sure that this is a correct approach.
